app/pages/capitalallowances/zeroEmissionGoodsVehicle/ZegvOnlyForSelfEmploymentPage.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages.capitalallowances.zeroEmissionGoodsVehicle
+
+import controllers.journeys.capitalallowances.zeroEmissionGoodsVehicle.routes
+import models.NormalMode
+import models.common._
+import models.database.UserAnswers
+import play.api.mvc.Call
+import queries.Settable
+import controllers.standard
+import pages.redirectOnBoolean
+
+object ZegvOnlyForSelfEmploymentPage extends ZegvBasePage[Boolean] {
+  override def toString: String = "zegvOnlyForSelfEmployment"
+
+  override val dependentPagesWhenYes: List[Settable[_]] =
+    List(
+      ZegvUseOutsideSEPercentagePage
+    )
```

Comment:
Needs to clear the ZegvUseOutsideSEPage as well

Reply:
done